source/common/router/config_impl.cc

```diff
@@ -46,6 +46,20 @@ RetryPolicyImpl::RetryPolicyImpl(const envoy::api::v2::RouteAction& config) {
   retry_on_ |= RetryStateImpl::parseRetryGrpcOn(config.retry_policy().retry_on());
 }
 
+CorsPolicyImpl::CorsPolicyImpl(const envoy::api::v2::CorsPolicy& config) {
+  for (const auto& origin : config.allow_origin()) {
+    allow_origin_.push_back(origin);
+  }
+  allow_methods_ = config.allow_methods();
+  allow_headers_ = config.allow_headers();
+  expose_headers_ = config.expose_headers();
+  max_age_ = config.max_age();
+  if (config.has_allow_credentials()) {
+    allow_credentials_.value(PROTOBUF_GET_WRAPPED_REQUIRED(config, allow_credentials));
```

Comment:
I'm a little confused as to why we need to make this optional. Can't you just default this to false?

Reply:
Premise: we check route first then fall back to vhost if not set.

If I default to false I cannot check if it's set or not. I could accept that it's false so the user would have to set true every time he adds a route config to change the headers or such. So I just made this behave the same. 
We could change it back to default false and document it. Might be unexpected though behavior though :)

Reply:
No that's fine, was just curious. Sounds good.

Reply:
I might even want to make expose headers the same. Because it could make sense to set it to an empty string. Have to confirm. The others make no sense to be empty I think. (Headers+methods+age)